src/main/java/tuna/command/AddDeadLineCommand.java

```diff
@@ -0,0 +1,50 @@
+package tuna.command;
+
+import java.util.Arrays;
+
+import tuna.Storage;
+import tuna.TaskList;
+import tuna.TunaException;
+import tuna.Ui;
+
+/**
+ * Represents an add Deadline Task command. An AddDeadLineCommand object contains the index of the /by command,
+ * task description and the deadline.
+ */
+public class AddDeadLineCommand extends Command {
+
+    /** Index of the /by command */
+    private int limit;
+    /** Description of the task */
+    private String taskDescription;
+    /** Deadline of the task */
+    private String by;
```

Comment:
can consider changing the 'by' variable name to 'deadline' to be more descriptive